modules/Chromecast/resources/sender/chromecast.js

```diff
@@ -69,10 +69,16 @@
          * @param reason
          */
         toggleCastButton: function ( isAvailable, reason ) {
+            var _this = this;
             this.log( "toggleCastButton: isAvailable=" + isAvailable + ", reason=" + reason );
             if ( isAvailable ) {
                 this.initializeCastApi();
                 this.show();
+                if (this.embedPlayer.getControlBarContainer().length < 1) {
+                    this.bind('onShowInterfaceComponents', function () {
```

Comment:
this is an event which is a side effect of the live core, don't see it used anywhere else.
Why not check the `embedPlayer.playerReadyFlag` and call show or wait to playerReady event according to it.

BTW, same goes to hide I guess.

Reply:
Done.
I used the onShowInterfaceComponents cause I saw the same event [here](https://github.com/kaltura/mwEmbed/blob/17765fee3badb1ea61e41542ec320d14d2339231/modules/KalturaSupport/resources/mw.KBaseComponent.js#L96-L100) and though it will be a better to use.